packages/eslint-plugin/tests/rules/prefer-readonly.test.ts

```diff
@@ -742,5 +742,75 @@ function ClassWithName<TBase extends new (...args: any[]) => {}>(Base: TBase) {
         },
       ],
     },
+    {
+      code: `
+        class Test {
+          private testObj = {
+            prop: '',
+          };
+
+          public test(): void {
+            this.testObj.prop = '';
+          }
+        }
+      `,
+      output: `
+        class Test {
+          private readonly testObj = {
+            prop: '',
+          };
+
+          public test(): void {
+            this.testObj.prop = '';
+          }
+        }
+      `,
+      errors: [
+        {
+          data: {
+            name: 'testObj',
+          },
+          line: 3,
+          messageId: 'preferReadonly',
+        },
+      ],
+    },
+    {
+      code: `
+        class TestObject {
+          public value: number;
+        }
+
+        class Test {
+          private testObj = new TestObject();
+
+          public test(): void {
+            this.testObj.value = 10;
+          }
+        }
+      `,
+      output: `
+        class TestObject {
+          public value: number;
+        }
+
+        class Test {
+          private readonly testObj = new TestObject();
+
+          public test(): void {
+            this.testObj.value = 10;
+          }
+        }
+      `,
+      errors: [
+        {
+          data: {
+            name: 'testObj',
+          },
+          line: 7,
+          messageId: 'preferReadonly',
+        },
+      ],
+    },
```

Comment:
A few more test cases to consider...

```ts
this.testObj.prop;
this.testObj?.prop;
this.testObj!.prop;
this.testObj!.prop = '';
this.testObj.prop.prop = '';
this.testObj?.prop.prop;
this.testObj?.prop?.prop;
this.testObj.prop?.prop;
this.testObj!.prop?.prop;
```

For any change to rule logic, it's generally good to try to add both `valid` and `invalid` test cases.

Reply:
@islandryu by "test cases" I do mean cases plural - we generally try to keep these separate, so that each test case tests one case. It's easy to miss incorrect or mismatched errors when they're all lumped together. Could you split them up please?